portal/static/js/profile.js

```diff
@@ -1939,157 +1948,126 @@
-                                            }
-                                        }
-                                        thisItem.parents(".pat-q").nextAll().fadeOut();
+                                        $("#biopsy_day").focus();
                                     }
-                                });
-                                var diag = $("#pca_diag_yes");
-                                if (diag.is(":checked")) {
-                                    diag.parents(".pat-q").nextAll().fadeIn();
                                 } else {
-                                    diag.parents(".pat-q").nextAll().fadeOut();
+                                    $("#biopsyDate").attr("skipped", "true");
+                                    $("#biopsyDate, #biopsy_day, #biopsy_month, #biopsy_year").val("");
+                                    $("#biopsyDateError").text("");
+                                    $("#biopsyDateContainer").hide();
+                                    arrQ.forEach(function(fieldName) {
+                                        var field = $("#patientQ input[name='" + fieldName + "']");
+                                        field.prop("checked", false);
+                                        field.attr("skipped", "true");
+                                    });
+                                    setTimeout(function() {
+                                        self.modules.tnthAjax.postClinical(userId, toCall, "false", thisItem.attr("data-status"), targetField);
+                                        self.modules.tnthAjax.postClinical(userId, "pca_diag", "false");
+                                        self.modules.tnthAjax.postClinical(userId, "pca_localized", "false");
+                                        self.modules.tnthAjax.deleteTreatment(userId);
+                                    }, 50);
+
                                 }
-                                if (self.currentUserId !== self.subjectId) {
-                                    if (!$("#patientQ input[type='radio']").is(":checked")) {
-                                        $("#patientQContainer").append("<span class='text-muted'>" + i18next.t("no answers provided") + "</span>");
-                                    }
+                            });
+                            self.__convertToNumericField($("#biopsy_day, #biopsy_year"));
+                            $("#biopsy_day, #biopsy_month, #biopsy_year").on("change", function() {
+                                var d = $("#biopsy_day"), m = $("#biopsy_month"), y = $("#biopsy_year");
+                                var isValid = self.modules.tnthDates.validateDateInputFields(m, d, y, "biopsyDateError");
+                                if (isValid) {
+                                    $("#biopsyDate").val(y.val()+"-"+m.val()+"-"+d.val());
```

Comment:
Might be nice to use a datetime library for formatting, can you use [`toLocaleDateString()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toLocaleDateString)?

Reply:
@ivan-c Ivan, I think so. I did try but LocaleDateString() is not returning datestring in ios short format (yyyy-mm-dd), at least it is not in chrome or firefox - a discussion about the inconsistencies:  https://stackoverflow.com/questions/27939773/tolocaledatestring-short-format.  Think want to leave it unchanged for now just to be safe.

Reply:
@achen2401 Thanks, it looks like there's not an easy way to get the formatting you want :(
I found this while looking, but it's only *slightly* better :/
```
var date = Date();
date.toISOString().split('T')[0];
```